internal/core/item_tree.rs

```diff
@@ -286,6 +286,108 @@ pub fn visit_item_tree<Base>(
     }
 }
 
+// Some helper to navigate the item_tree:
+
+/// This is a local item tree of one component with some helpful functions to navigate the tree.
+pub struct ItemTree<'a, T> {
+    item_tree: &'a [ItemTreeNode<T>],
+}
+
+impl<'a, T> ItemTree<'a, T> {
+    /// Cretae a new `ItemTree` from its raw data.
+    pub fn new(item_tree: &'a [ItemTreeNode<T>]) -> Self {
+        Self { item_tree }
+    }
+
+    /// Get the parent of a node, returns `None` if this is the root node of this item tree.
+    pub fn parent(&self, index: usize) -> Option<usize> {
+        if index >= self.item_tree.len() || index == 0 {
+            None
+        } else {
+            match &self.item_tree[index] {
+                ItemTreeNode::Item { parent_index, .. } => Some(*parent_index as _),
+                ItemTreeNode::DynamicTree { parent_index, .. } => Some(*parent_index as _),
+            }
+        }
```

Comment:
```suggestion
        (index < self.item_tree.len() && index != 0).then(|| self.item_tree[index].parent_index())
```
There is already a parent_index function.